kaldb/src/main/java/com/slack/kaldb/clusterManager/ReplicaRestoreService.java

```diff
@@ -0,0 +1,160 @@
+package com.slack.kaldb.clusterManager;
+
+import static com.slack.kaldb.clusterManager.ReplicaCreationService.replicaMetadataFromSnapshotId;
+
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.slack.kaldb.metadata.replica.ReplicaMetadata;
+import com.slack.kaldb.metadata.replica.ReplicaMetadataStore;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadata;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+import javax.naming.SizeLimitExceededException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Scheduled service responsible for queueing and creating replica snapshots */
+public class ReplicaRestoreService extends AbstractScheduledService {
+  private ScheduledFuture<?> pendingTask;
+  private final KaldbConfigs.ManagerConfig managerConfig;
+  private final KaldbConfigs.ManagerConfig.ReplicaRestoreServiceConfig replicaRestoreServiceConfig;
+  private final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
```

Comment:
This probably should have a thread factory provided to it - I know we're not super consistent with this in the codebase but it makes troubleshooting a bit easier since the thread name is printed into the logs.

Check out ReadOnlyChunkImpl:91 for an example of us passing a named thread factory to this